front-end/src/context/tokenContext.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, useContext, useState, useEffect} from "react"
+import {baseURL, client} from "../config/AxiosConfig";
+import Cookies from 'universal-cookie';
+
+export type userTokenContent = {
+  user_token: string
+  setUserToken:(c: string) => void
+}
+
+export const MyTokenContext = createContext<userTokenContent>({
+    user_token: '',
+    setUserToken: () => {},
+})
+
+export default function Provider({children}:any) {
+    const cookies = new Cookies();
+    const [user_token, setUserToken] = useState<string>(cookies.get('user_token'))
+    const headers = {'Authorization': 'Bearer '+user_token}
+
+    const isTokenExpired = async (user_token:string) => {
```

Comment:
Error handling: The code is not currently handling errors very well. For example, if the network request to check the token status fails, the function will return undefined, which could cause issues downstream. It would be a good idea to add proper error handling using try-catch blocks or error handling functions.